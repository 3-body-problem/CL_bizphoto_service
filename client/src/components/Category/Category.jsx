import React from 'react';
import PropTypes from 'prop-types';

const Category = ({ price, category }) => (
  <div className="price-category">
    <span className="bullet">
      <span className="price-range">{price}</span>
    </span>
    <span className="category-list">
      {` ${category.join(', ')} `}
    </span>
    <a
      href="https://www.yelp.com"
      className="chiclet-link chiclet-link-text"
    >
      Edit
    </a>
  </div>
);

Category.propTypes = {
  price: PropTypes.string.isRequired,
  category: PropTypes.array.isRequired,
};
export default Category;
