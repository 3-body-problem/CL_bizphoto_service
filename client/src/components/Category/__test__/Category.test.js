import { mount } from 'enzyme';
import React from 'react';
import Category from '../Category';

describe('Render Category', () => {
  it('render correctly Category component', () => {
    const props = {
      price: '$$$',
      category: ['Steakhouses', 'Wine Bars'],
    };
    const CategoryComponent = mount(<Category {...props} />);
    expect(CategoryComponent).toMatchSnapshot();
  });
  it('check props with html tags', () => {
    const props = {
      price: '$',
      category: ['Chinese', 'Wine Bars', 'American'],
    };
    const CategoryComponent = mount(<Category {...props} />);
    expect(CategoryComponent.find('span.price-range').text()).toEqual('$');
    expect(CategoryComponent.find('span.category-list').text()).toEqual(' Chinese, Wine Bars, American ');
  });
});
