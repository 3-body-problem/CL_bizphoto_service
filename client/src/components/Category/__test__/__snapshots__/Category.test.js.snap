// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Category render correctly Category component 1`] = `
<Category
  category={
    Array [
      "Steakhouses",
      "Wine Bars",
    ]
  }
  price="$$$"
>
  <div
    className="price-category"
  >
    <span
      className="bullet"
    >
      <span
        className="price-range"
      >
        $$$
      </span>
    </span>
    <span
      className="category-list"
    >
       Steakhouses, Wine Bars 
    </span>
    <a
      className="chiclet-link chiclet-link-text"
      href="https://www.yelp.com"
    >
      Edit
    </a>
  </div>
</Category>
`;
